sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -913,6 +933,21 @@ private Schema fetchAvroSchema(String schemaRegistryUrl, String subject) {
       return avroSchema;
     }
 
+    static SchemaRegistryClient getSchemaRegistryClient(
+        String schemaRegistryUrl,
+        SerializableFunction<String, SchemaRegistryClient> schemaRegistryClientFactoryFn) {
+      SchemaRegistryClient registryClient;
+      if (schemaRegistryUrl == null) {
```

Comment:
`checkArgument` better, not having a URL to get the client is an Exception condition.